sstables/sstables.cc

```diff
@@ -3145,16 +3146,21 @@ future<bool> sstable::has_partition_key(const utils::hashed_key& hk, const dht::
     std::exception_ptr ex;
     auto sem = reader_concurrency_semaphore(reader_concurrency_semaphore::no_limits{}, "sstables::has_partition_key()",
             reader_concurrency_semaphore::register_metrics::no);
```

Comment:
> We might as well ignore the index in all cases and just check
> that a regular data reader returns a non-empty result.
> `sstable::has_partition_key` is only used in the
> `column_family/sstables/by_key` REST API call that nobody
> uses anyway, no point in trying to microoptimize it.

Please check that we have a test for by_key. I do see we have tests that use it, but that's not the same thing. Either test.py or dtest are fine.